Maintenance: automate header guards 2/3

kinkie opened the pull request:
In preparation for merging the automated header-guards maintenance
script, merge the changes brought by that script.

rousskov reviewed (approved):
I have adjusted PR description (similar to https://github.com/squid-cache/squid/pull/1654#pullrequestreview-1851404620). I am approving this PR without checking details, but it has to wait for #1564 to merge: AFAICT, without that PR, this PR fails our CI tests (including, but not limited to, GitHub CodeQL tests).

yadij commented:
@kinkie, why remove the v6 backport label?

kinkie commented:
> @kinkie, why remove the v6 backport label?

These are the ones that are automatically generated. If we added, removed, or renamed a file since v6 was forked, this merge will fail. I'd rather regenerate and have a v6-specific equivalent PR
